src/wwwroot/js/genpage/gentab/outputhistory.js

```diff
@@ -98,6 +98,31 @@ function buttonsForImage(fullsrc, src, metadata) {
                 if (!uiImprover.lastShift && getUserSetting('ui.checkifsurebeforedelete', true) && !confirm('Are you sure you want to delete this image?\nHold shift to bypass.')) {
                     return;
                 }
+                let wasFullView = imageFullView && imageFullView.isOpen && imageFullView.isOpen();
+                let curImg = document.getElementById('current_image_img');
+                let keepOpen = false;
+                if (wasFullView && curImg) {
+                    if (curImg.dataset.batch_id == 'history' && lastHistoryImageDiv && lastHistoryImageDiv.parentElement) {
```

Comment:
this code does not need to consider anything specific to batch/history stuff, just call `shiftToNextImagePreview`